libimage/image.go

```diff
@@ -112,6 +113,13 @@ func (i *Image) Names() []string {
 	return i.storageImage.Names
 }
 
+// cleanNames removes all names that are not in the resolvedNames list.
+func (i *Image) cleanNames(resolvedNames []string) {
```

Comment:
“clean” is not quite what this does, and “resolved” is not important for the purpose of this function.

(But let’s settle the data passing mechanism first.)